marimo/_runtime/dataflow.py

```diff
@@ -96,11 +96,13 @@ def get_referring_cells(
         """
         if language == "sql":
             # For SQL, only return SQL cells that reference the name
-            return {
-                cid
-                for cid, cell in self.cells.items()
-                if name in cell.refs and cell.language == "sql"
-            }
+            cells = set()
+            for cid, cell in self.cells.items():
+                if cell.language == "sql":
+                    for ref in cell.refs:
+                        if name == ref or ("." in ref and name in ref):
+                            cells.add(cid)
+            return cells
```

Comment:
btw, the current bugfix referenced in the PR description is in the `else` statement im pretty sure. the failing code is `python`